### PR TITLE
Improve performance by wrapping event handlers in useCallback in rjsf/core

### PR DESCRIPTION
**Pull request by jimmycallin**
### Reasons for making this change

Improve performance by wrapping event handlers in useCallback in rjsf/core

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by heath-freenome** [comment]
Can you also update the `templates` directory too? And perhaps the widgets and template in all the themes (in other PRs)

File: packages/core/src/components/widgets/CheckboxWidget.tsx [hunk 1]
```diff
@@ -30,14 +30,23 @@ function CheckboxWidget<T = any, F = any>({
...
+      onChange(event.target.checked),
+    [onChange]
+  );
 
-  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) =>
-    onBlur(id, event.target.checked);
+  const handleBlur = useCallback(
+    (event: React.FocusEvent<HTMLInputElement>) =>
+      onBlur(id, event.target.checked),
+    []
```
Comment (heath-freenome) [inline packages/core/src/components/widgets/CheckboxWidget.tsx:42]: `[onBlur, id]`?

Comment (jimmycallin) [reply packages/core/src/components/widgets/CheckboxWidget.tsx]: nice catch

**Comment by jimmycallin** [comment]
> Can you also update the `templates` directory? And perhaps the widgets and templates in all the themes (in other PRs)

i only found one applicable file in the templates directory - BaseInputTemplate. please verify i didn't miss anything. :)

i'll see if i get around to update the themes as well.

**Review by heath-freenome — APPROVED** [review]

**Comment by heath-freenome** [comment]
> > Can you also update the `templates` directory? And perhaps the widgets and templates in all the themes (in other PRs)
> 
> i only found one applicable file in the templates directory - BaseInputTemplate. please verify i didn't miss anything. :)

Looks good!
> 
> i'll see if i get around to update the themes as well.
That would be great!

**Pull request merged**